activerecord/lib/active_record/persistence.rb

```diff
@@ -249,8 +249,8 @@ def insert_all!(attributes, returning: nil, record_timestamps: nil)
       # go through Active Record's type casting and serialization.
       #
       # See #upsert_all for documentation.
-      def upsert(attributes, on_duplicate: :update, returning: nil, unique_by: nil, record_timestamps: nil)
-        upsert_all([ attributes ], on_duplicate: on_duplicate, returning: returning, unique_by: unique_by, record_timestamps: record_timestamps)
+      def upsert(attributes, **kwargs)
+        upsert_all([ attributes ], **kwargs)
```

Comment:
@gregmolnar  Nice! i'm glad my comment helped improve ActiveRecord😄

Reply:
@chaadow thanks for letting me know about this discrepancy!